jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpStatus.java

```diff
@@ -383,7 +383,7 @@ public static boolean isSuccess(int code)
      */
     public static boolean isRedirection(int code)
     {
-        return ((300 <= code) && (code <= 399));
+        return ((300 <= code) && (code <= 399) && code != 304);
```

Comment:
Seems like you added this extra 304 status code hack here just to avoid producing a response body.
Seems like the wrong place for this special case.

Also, with 304 we should never produce trailers.
Perhaps this needs to be somewhere else?

Reply:
Strictly per RFC, 304 is a redirection status code.

We have a similar issue for informational and interim status codes, and ended up having 2 methods.
Perhaps add a `isRedirect(int)` or `isLocationRedirect(int)` and exclude 304 in this new method?

Reply:
@joakime No.  This is to prevent 304 from being used with a sendRedirect.  But 304 is a redirect, so I will do it elsewhere.